harmonica/equivalent_layer/harmonic.py

```diff
@@ -186,23 +197,197 @@ def jacobian(
         jacobian : 2D array
             The (n_data, n_points) Jacobian matrix.
         """
+        # Compute Jacobian matrix
         n_data = coordinates[0].size
         n_points = points[0].size
         jac = np.zeros((n_data, n_points), dtype=dtype)
-        jacobian_numba(coordinates, points, jac)
+        jacobian_numba(coordinates, points, jac, self.greens_function)
         return jac
 
 
+class EQLHarmonicSpherical(EQLHarmonic):
+    r"""
+    Equivalent-layer for generic harmonic functions in spherical coordinates
+
+    This equivalent layer can be used for:
+
+    * Spherical coordinates (geographic coordinates must be converted before
+      use)
+    * Regional or global data where Earth's curvature must be taken into
+      account
+    * Gravity and magnetic data (including derivatives)
+    * Single data types
+    * Interpolation
+    * Upward continuation
+    * Finite-difference based derivative calculations
+
+    It cannot be used for:
+
+    * Joint inversion of multiple data types (e.g., gravity + gravity
+      gradients)
+    * Reduction to the pole of magnetic total field anomaly data
+    * Analytical derivative calculations
+
+    Point sources are located beneath the observed potential-field measurement
+    points by default [Cooper2000]_. Custom source locations can be used by
+    specifying the *points* argument. Coefficients associated with each point
+    source are estimated through linear least-squares with damping (Tikhonov
+    0th order) regularization.
+
+    The Green's function for point mass effects used is the inverse Cartesian
+    distance between the grid coordinates and the point source:
+
+    .. math::
+
+        \phi(\bar{x}, \bar{x}') = \frac{1}{||\bar{x} - \bar{x}'||}
+
+    where :math:`\bar{x}` and :math:`\bar{x}'` are the coordinate vectors of
+    the observation point and the source, respectively.
+
+    Parameters
+    ----------
+    damping : None or float
+        The positive damping regularization parameter. Controls how much
+        smoothness is imposed on the estimated coefficients.
+        If None, no regularization is used.
+    points : None or list of arrays (optional)
+        List containing the coordinates of the point sources used as the
+        equivalent layer. Coordinates are assumed to be in the following order:
+        (``longitude``, ``latitude``, ``radius``). Both ``longitude`` and
```

Comment:
Should we differentiate latitude from *spherical latitude* here?

Reply:
Maybe not since we say it's spherical in the docstring and we didn't do this of the forward modeling functions, right?

Reply:
I agree with that. If we decide to change it to spherical latitude here, we must do it elsewhere. But I don't think this is critical. The `radius` coordinate next to it would make it sufficiently clear to any usual user of Harmonica that it's asking spherical coordinates. And for any newcomer I think we should make this very clear on the _Page about coordinate systems_ (#68).

Reply:
Right, that's probably better